crates/nu-command/src/network/http/client.rs

```diff
@@ -561,14 +560,12 @@ fn transform_response_using_content_type(
     resp: Response,
     content_type: &str,
 ) -> Result<PipelineData, ShellError> {
-    let regex = Regex::new("\"").expect("Failed to create regex");
-    let content_type_trim = regex.replace_all(content_type, "").to_string();
-
-    let content_type = mime::Mime::from_str(&content_type_trim).map_err(|err| {
-        LabeledError::new(err.to_string())
-            .with_help("given unknown MIME type, or error parsing MIME type")
-            .with_label(format!("MIME type unknown: {content_type_trim}"), span)
-    })?;
+    let content_type = mime::Mime::from_str(content_type)
+        // there are invalid content types in the wild, so we try to recover
+        // Example: `Content-Type: "text/plain"; charset="utf8"` (note the quotes)
+        .or_else(|_| mime::Mime::from_str(&content_type.replace('"', "")))
```

Comment:
This probably ok.. though maybe we should wrap this call. From my experience, the mime library doesn't always return the correct mime type. For instance, the mime library returns text/x-yaml for mime, which is no longer correct as of RFC-9512. The correct mime type is application/yaml. I had to handle this in filesystem/open.rs:

```rust
fn detect_content_type(extension: &str) -> Option<String> {
    // This will allow the overriding of metadata to be consistent with
    // the content type
    match extension {
        // Per RFC-9512, application/yaml should be used
        "yaml" | "yml" => Some("application/yaml".to_string()),
        _ => mime_guess::from_ext(extension)
            .first()
            .map(|mime| mime.to_string()),
    }
}
```

Reply:
> For instance, the mime library returns text/x-yaml for mime, which is no longer correct as of RFC-9512. 

That's not applicable here, as we're just parsing the `Content-Type` header instead of any kind of content sniffing.